src/batch_norm_api.cpp

```diff
@@ -264,10 +264,14 @@ miopenBatchNormalizationForwardInference_V2(miopenHandle_t handle,
             (size == 5) ? miopen::BuildReshaped4DTensorDescriptor(miopen::deref(yDesc))
                         : miopen::deref(yDesc),
             DataCast(y),
-            miopen::deref(scaleDesc),
-            miopen::deref(biasDesc),
-            miopen::deref(estMeanDesc),
-            miopen::deref(estVarianceDesc),
+            (size == 5) ? miopen::BuildReshaped4DTensorDescriptor(miopen::deref(scaleDesc))
+                        : miopen::deref(scaleDesc),
+            (size == 5) ? miopen::BuildReshaped4DTensorDescriptor(miopen::deref(biasDesc))
+                        : miopen::deref(biasDesc),
+            (size == 5) ? miopen::BuildReshaped4DTensorDescriptor(miopen::deref(estMeanDesc))
+                        : miopen::deref(estMeanDesc),
+            (size == 5) ? miopen::BuildReshaped4DTensorDescriptor(miopen::deref(estVarianceDesc))
+                        : miopen::deref(estVarianceDesc),
```

Comment:
```C++
auto ReshapeIfNeeded = [size](const auto desc)
{
    return (size == 5) ? miopen::BuildReshaped4DTensorDescriptor(miopen::deref(desc))
                       : miopen::deref(desc);
};
```

```suggestion
           ReshapeIfNeeded(scaleDesc),
           ReshapeIfNeeded(biasDesc),
           ReshapeIfNeeded(estMeanDesc),
           ReshapeIfNeeded(estVarianceDesc)
```
+ the same for `xDesc` and `yDesc` and for `miopenBatchNormalizationForwardTraining_V2`